Pangolin/pangolearn failed, data column mismatch?

Hi @aineniamh 
I don't understand the error, could you help

```
pangolin pangolin -pv
pangoLEARN 2020-08-29_3
pangolin -lv
lineages 2020-05-19
pangolin -v
pangolin 2.0.8
python --version
Python 3.6.10
```


```
Looking in /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data for data files...

Data files found
Trained model:  /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/decisionTree_v1.joblib
Header file:    /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/decisionTreeHeaders_v1.joblib
Lineages csv:   /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/lineages.metadata.csv
Job counts:
        count   jobs
        1       add_failed_seqs
        1       all
        1       datafunk_trim_and_pad
        1       minimap2_check_distance
        1       minimap2_to_reference
        1       pangolearn
        1       parse_paf
        7

        minimap2 -x asm5 /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangolin/data/reference.fasta /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/query.post_qc.fasta -o /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/reference_mapped.paf &> /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/logs/minimap2_check.log

Job counts:
        count   jobs
        1       parse_paf
        1

        minimap2 -a -x asm5 /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangolin/data/reference.fasta /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/mappable.fasta -o /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/reference_mapped.sam &> /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/logs/minimap2_sam.log


        datafunk sam_2_fasta           -s /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/reference_mapped.sam           -r /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangolin/data/reference.fasta           -o /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/post_qc_query.aligned.fasta           -t [265:29674]           --pad           --log-inserts


        pangolearn.py --header-file /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/decisionTreeHeaders_v1.joblib --model-file /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/decisionTree_v1.joblib --reference-file /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangolin/data/reference.fasta --fasta /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/post_qc_query.aligned.fasta -o /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/lineage_report.pass_qc.csv

loading model 10/02/2020, 09:09:35
processing block of 1001 sequences 10/02/2020, 09:09:51
Traceback (most recent call last):
  File "/home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pandas/core/internals/construction.py", line 497, in _list_to_arrays
    content, columns, dtype=dtype, coerce_float=coerce_float
  File "/home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pandas/core/internals/construction.py", line 581, in _convert_object_array
    f"{len(columns)} columns passed, passed data had "
AssertionError: 15818 columns passed, passed data had 29903 columns

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/bin/opt/mc3/envs/pangolin/bin/pangolearn.py", line 125, in <module>
    df = pd.DataFrame(seqList, columns=indiciesToKeep)
  File "/home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pandas/core/frame.py", line 474, in __init__
    arrays, columns = to_arrays(data, columns, dtype=dtype)
  File "/home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pandas/core/internals/construction.py", line 461, in to_arrays
    return _list_to_arrays(data, columns, coerce_float=coerce_float, dtype=dtype)
  File "/home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pandas/core/internals/construction.py", line 500, in _list_to_arrays
    raise ValueError(e) from e
ValueError: 15818 columns passed, passed data had 29903 columns
[Fri Oct  2 09:09:51 2020]
Error in rule pangolearn:
    jobid: 2
    output: /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/lineage_report.pass_qc.csv
    shell:

        pangolearn.py --header-file /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/decisionTreeHeaders_v1.joblib --model-file /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangoLEARN/data/decisionTree_v1.joblib --reference-file /home/bin/opt/mc3/envs/pangolin/lib/python3.6/site-packages/pangolin/data/reference.fasta --fasta /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/post_qc_query.aligned.fasta -o /ncov-aau/results/Denmark/pangtmp/tmpcojya8l_/lineage_report.pass_qc.csv

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

--- a/pangolin/scripts/pangolearn.py
+++ b/pangolin/scripts/pangolearn.py
@@ -91,8 +91,8 @@ def readInAndFormatData(sequencesFile, indiciesToKeep, blockSize=1000):
 					currentSeq = currentSeq + line
 
 			if len(seqList) == blockSize:
-				idList.append(referenceId)
-				seqList.append(referenceSeq)
+				# idList.append(referenceId)
+				# seqList.append(referenceSeq)
 				yield idList, seqList
 				idList = []
 				seqList = []
@@ -127,7 +127,7 @@ for idList, seqList in readInAndFormatData(args.sequences_file, indiciesToKeep):
 	# possible nucleotide symbols
 	categories = ['A', 'C', 'G', 'T', '-']
 
-	# add extra rows to ensure all of the categories are represented, as otherwise 
+	# add extra rows to ensure all of the categories are represented, as otherwise
 	# not enough columns will be created when we call get_dummies
 	for i in categories:
 		line = [i] * len(indiciesToKeep)

